proxy/http/HttpSM.cc

```diff
@@ -6560,6 +6559,12 @@ HttpSM::kill_this()
   reentrancy_count--;
   ink_release_assert(reentrancy_count == 0);
 
+  // Delay the close of the user agent session, so the close session
+  // occurs after the close transaction
+  if (ua_session != NULL)
```

Comment:
The existing code is pretty eager to do_io_close() on the VC in error cases anyway.